test/sql/aws_secret_validation.test

```diff
@@ -0,0 +1,130 @@
+# name: test/sql/settings/aws_secret_validation_present.test
+# description: Test that SET secret_validation works
+# group: [sql]
+
+require aws
+
+require httpfs
+
+require-env S3_TEST_SERVER_AVAILABLE 1
+
+# Without params, 'exists' check must pass using secret_directory
+statement ok
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain
+);
+
+# Explicit 'exists' check must pass using secret_directory
+statement ok
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain,
+    VALIDATION 'exists'
+);
+
+# Explicit 'none' check must also pass
+statement ok
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain,
+    VALIDATION 'none'
+);
+
+#### Confirm also with legit profile
+
+# Without params, 'exists' check must pass using secret_directory
+statement ok
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain,
+	PROFILE 'minio-testing-2'
+);
+
+# Explicit 'exists' check must pass using secret_directory
+statement ok
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain,
+	PROFILE 'minio-testing-2',
+    VALIDATION 'exists'
+);
+
+# Explicit 'none' check must also pass
+statement ok
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain,
+	PROFILE 'minio-testing-2',
+    VALIDATION 'none'
+);
+
+
+#### Use non existent profile to force no available credentials
+
+# bad validation param
+statement ok
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain,
+	PROFILE 'minio-testing-empty',
+    VALIDATION 'none'
+);
+
+# validation param specified correctly, no credential found to exist
+statement error
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain,
+	PROFILE 'minio-testing-empty',
+    VALIDATION 'exists'
+);
+----
+<REGEX>:.*Invalid Input Error: Failed to create secret.*
+
+# validation param specified correctly, no credential found to exist
+statement error
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain,
+	PROFILE 'minio-testing-empty',
+    VALIDATION 'nonsense'
+);
+----
+<REGEX>:.*Invalid Configuration Error: Unknown AWS VALIDATION option.*
+
+
+#### confirm the same with non-existent profile
+
+# bad validation param
+statement ok
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain,
+	PROFILE 'no-such-profile',
+    VALIDATION 'none'
+);
+
+# validation param specified correctly, no credential found to exist
+statement error
+CREATE OR REPLACE SECRET aws (
+    TYPE S3,
+    PROVIDER credential_chain,
+	PROFILE 'no-such-profile',
+    VALIDATION 'exists'
+);
+----
+<REGEX>:.*Invalid Configuration Error: Failed to load profile.*
```

Comment:
again here: I think referencing the relationship to the VALIDATION option here would be nice?